pkg/backup/main.go

```diff
@@ -136,14 +143,33 @@ func DoBackupRestore(backupURL string, toFile string, restoreObj *replica.Restor
 	backupURL = util.UnescapeURL(backupURL)
 	log.Debugf("Start restoring from %v into snapshot %v", backupURL, toFile)
 
+	retryCount := 0
+	bf := &backoff.Backoff{
+		Min:    backoffMinInterval,
+		Max:    backoffMaxInterval,
+		Factor: backoffFactor,
+		Jitter: false,
+	}
+
 	config := &backupstore.DeltaRestoreConfig{
 		BackupURL: backupURL,
 		DeltaOps:  restoreObj,
 		Filename:  toFile,
 	}
 
-	if err := backupstore.RestoreDeltaBlockBackup(config); err != nil {
-		return err
+	for {
```

Comment:
I don't think we can add a wait & retry loop here. This will block the restore call then lead to timeout errors.

Reply:
Yes, so It has just 5 times to retry and timeout in one minute.
I should turn to `Async part` to handle the lock acquisition failed, right?

Reply:
or Could I check the getting lock failed here [sync() and refresh()](https://github.com/longhorn/longhorn-manager/blob/master/controller/engine_controller.go#L727) and retry?